test/recipeModuleManager.test.js

```diff
@@ -45,4 +57,674 @@ describe(`recipeModuleManagerTest: ${printPath("[test/recipeModuleManager.test.j
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                websiteDomain: "supertokens.io",
+            },
+            recipeList: [TestRecipe1.init()],
+        });
+
+        const app = express();
+
+        app.use(ST.middleware());
+
+        app.post("/auth/default-route-disabled", async (req, res) => {
+            res.status(200).json({ message: "user defined api" });
+        });
+
+        app.use(ST.errorHandler());
+
+        let r1 = await new Promise((resolve) =>
+            request(app)
+                .post("/auth/default-route-disabled")
+                .expect(200)
+                .end((err, res) => {
+                    if (err) {
+                        resolve(undefined);
+                    } else {
+                        resolve(res.body.message);
+                    }
+                })
+        );
+        assert(r1 === "user defined api");
+    });
+
+    // TODO: If an error handler in a recipe throws an error, that error next to go to the user's error handler
+    it("test if the error handler in a recipe throws an error, it goes to the user's error handler", async function () {
+        await startST();
+
+        ST.init({
+            supertokens: {
+                connectionURI: "http://localhost:8080",
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                websiteDomain: "supertokens.io",
+            },
+            recipeList: [TestRecipe1.init()],
+        });
+
+        const app = express();
+
+        app.use(ST.middleware());
+        app.use(ST.errorHandler());
+        app.use((err, request, response, next) => {
+            if (err.message === "error from inside recipe error handler") {
+                response.status(200).send("user error handler");
+            } else {
+                response.status(500).send("invalid error");
+            }
+        });
+
+        let r1 = await new Promise((resolve) =>
+            request(app)
+                .post("/auth/error/throw-error")
+                .expect(200)
+                .end((err, res) => {
+                    if (err) {
+                        resolve(undefined);
+                    } else {
+                        resolve(res.text);
+                    }
+                })
+        );
+        assert(r1 === "user error handler");
+    });
+
+    // TODO: Test getAllCORSHeaders
+    it("test the getAllCORSHeaders function", async function () {
+        await startST();
+
+        ST.init({
+            supertokens: {
+                connectionURI: "http://localhost:8080",
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                websiteDomain: "supertokens.io",
+            },
+            recipeList: [TestRecipe1.init()],
```

Comment:
Do this test with multiple recipes. Each recipe should add something unique to the cors list, and make them both add the same value as well (to test that the final result doesn't have duplicates)